drivers/counter/counter_smartbond_timer.c

```diff
@@ -59,16 +59,22 @@ struct counter_smartbond_config {
 	LOG_INSTANCE_PTR_DECLARE(log);
 };
 
-#if defined(CONFIG_PM_DEVICE) || defined(CONFIG_PM_DEVICE_RUNTIME)
-static void counter_smartbond_pm_policy_state_lock_get(struct counter_smartbond_data *data)
+#if defined(CONFIG_PM_DEVICE)
+static void counter_smartbond_pm_policy_state_lock_get(const struct device *dev)
 {
+	struct counter_smartbond_data *data = dev->data;
+
 	if (!atomic_test_and_set_bit(data->pm_policy_state_flag, 0)) {
 		pm_policy_state_lock_get(PM_STATE_STANDBY, PM_ALL_SUBSTATES);
 	}
+	pm_device_runtime_get(dev);
```

Comment:
all these reference count, e.g `atomic_test_and_set_bit(data->pm_policy_state_flag, 0)`, is because you have unbalanced calls for `counter_smartbond_pm_policy_state_lock_get` ? if so, wouldn't the device reference count gets unbalanced ?

Reply:
counter driver there are no unbalanced calls. We could drop `atomic_test_and_set_bit(data->pm_policy_state_flag, 0)`.
We had unbalanced calls in UART and I2C drivers so we followed a different implementation there